tests/functional/test_report_only_no_errors.py

```diff
@@ -59,11 +59,15 @@ class Secret:
 
     bot_name = os.environ.get("BOT_NAME")
     bot_token = os.environ.get("BOT_TOKEN")
-    if not bot_token:
+    if not bot_name and not bot_token:
         bot_name = "github-actions[bot]"
         bot_token = os.environ.get("GITHUB_TOKEN")
         if not bot_token:
             raise Exception("BOT_TOKEN environment variable not defined")
+    elif not bot_name:
+        raise Exception("BOT_TOKEN set but BOT_NAME not specified")
+    elif not bot_token:
+        raise Exception("BOT_NAME set but BOT_TOKEN not specified")
```

Comment:
this is screaming for a function to set or validate bot_token in common

Reply:
Sorry I don't quite get what that means

Reply:
oh, every test module had the same logic check for diff combinations of bot_name and/or bot_token can you pull that out to a function that you call on all test modules. Search for 'raise Exception("BOT_TOKEN set but BOT_NAME not specified")'

Reply:
Thanks for the clarification.
Yea that sounds good to me, will do refactoring on this